src/test/integration/IntegrationDeployer.t.sol

```diff
@@ -0,0 +1,489 @@
+                eigenLayerReputedMultisig,
+                pauserRegistry,
+                0 // initialPausedStatus
+            )
+        );
+        // EigenPodManager
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(eigenPodManager))),
+            address(eigenPodManagerImplementation),
+            abi.encodeWithSelector(
+                EigenPodManager.initialize.selector,
+                type(uint256).max, // maxPods
+                address(beaconChainOracle),
+                eigenLayerReputedMultisig, // initialOwner
+                pauserRegistry,
+                0 // initialPausedStatus
+            )
+        );
+        // Delayed Withdrawal Router
+        uint256 withdrawalDelayBlocks = 7 days / 12 seconds;
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(delayedWithdrawalRouter))),
+            address(delayedWithdrawalRouterImplementation),
+            abi.encodeWithSelector(
+                DelayedWithdrawalRouter.initialize.selector,
+                eigenLayerReputedMultisig, // initialOwner
+                pauserRegistry,
+                0, // initialPausedStatus
+                withdrawalDelayBlocks
+            )
+        );
+
+        // Create base strategy implementation and deploy a few strategies
+        baseStrategyImplementation = new StrategyBase(strategyManager);
+
+        _newStrategyAndToken("Strategy1Token", "str1", 10e50, address(this)); // initialSupply, owner
+        _newStrategyAndToken("Strategy2Token", "str2", 10e50, address(this)); // initialSupply, owner
+        _newStrategyAndToken("Strategy3Token", "str3", 10e50, address(this)); // initialSupply, owner
+
+        ethStrats.push(BEACONCHAIN_ETH_STRAT);
+        mixedStrats.push(BEACONCHAIN_ETH_STRAT);
+
+        global = new Global();
+    }
+
+    /// @dev Deploy a strategy and its underlying token, push to global lists of tokens/strategies, and whitelist
+    /// strategy in strategyManager
+    function _newStrategyAndToken(string memory tokenName, string memory tokenSymbol, uint initialSupply, address owner) internal {
+        IERC20 underlyingToken = new ERC20PresetFixedSupply(tokenName, tokenSymbol, initialSupply, owner); 
+        StrategyBase strategy = StrategyBase(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(baseStrategyImplementation),
+                    address(eigenLayerProxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, underlyingToken, pauserRegistry)
+                )
+            )
+        );
+
+        // Push to list of tokens/strategies
+        _strategies.push(strategy);
+
+        // Whitelist strategy
+        IStrategy[] memory strategies = new IStrategy[](1);
+        strategies[0] = strategy;
+        strategyManager.addStrategiesToDepositWhitelist(strategies);
```

Comment:
gonna need a `cheats.prank(strategyManager.strategyWhitelister())` just before the `strategyManager.addStrategiesToDepositWhitelist` call

Reply:
[6ef08a6](https://github.com/Layr-Labs/eigenlayer-contracts/pull/335/commits/6ef08a676fe70389631c4627a9aa2d825a660884)